core/lib/config/src/configs/external_price_api_client.rs

```diff
@@ -0,0 +1,34 @@
+use std::time::Duration;
+
+use serde::Deserialize;
+use url::Url;
+
+pub const DEFAULT_TIMEOUT_MS: u64 = 10_000;
+pub const DEFAULT_COINGECKO_API_URL: &str = "https://pro-api.coingecko.com";
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ExternalPriceApiClientConfig {
+    #[serde(default = "ExternalPriceApiClientConfig::default_base_url")]
+    pub base_url: String,
+    pub api_key: Option<String>,
+    #[serde(default = "ExternalPriceApiClientConfig::default_timeout")]
+    pub client_timeout_ms: u64,
+}
+
+impl ExternalPriceApiClientConfig {
+    fn default_timeout() -> u64 {
+        DEFAULT_TIMEOUT_MS
+    }
+
+    fn default_base_url() -> String {
+        DEFAULT_COINGECKO_API_URL.to_string()
+    }
```

Comment:
...e.g. it's rather confusing; if you don't know what API you are going to talk to, the "default" URL doesn't make much sense.